src/redux/selectedThemeSlice.ts

```diff
@@ -0,0 +1,38 @@
+import { Theme, createTheme } from '@mui/material';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { themes } from '../utils/themes';
+
+const OSTheme = window
+  .matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+
+localStorage.setItem('OSTheme', OSTheme);
+
+const getTheme = (themeName: string): Theme => {
+  const theme = themes.find((foundTheme) => foundTheme.themeName === themeName)
+    || { themeName: '', themeObject: createTheme() };
+
+  return createTheme(theme.themeObject);
+};
+
+const themeNameFromLocalStorage = localStorage.getItem('theme');
+
+const initialState: Theme = themeNameFromLocalStorage
+  ? getTheme(themeNameFromLocalStorage)
+  : getTheme(OSTheme);
+
+const selectedThemeSlice = createSlice({
```

Comment:
probably better to call this slice just 'themeSlice', because you can have many states related to the theme there.